cmd/quadlet/main.go

```diff
@@ -107,31 +107,33 @@ func Debugf(format string, a ...interface{}) {
 // For system generators these are in /usr/share/containers/systemd (for distro files)
 // and /etc/containers/systemd (for sysadmin files).
 // For user generators these can live in $XDG_RUNTIME_DIR/containers/systemd, /etc/containers/systemd/users, /etc/containers/systemd/users/$UID, and $XDG_CONFIG_HOME/containers/systemd
-func getUnitDirs(rootless bool) []string {
-	// Allow overriding source dir, this is mainly for the CI tests
-	if varExist, dirs := getDirsFromEnv(); varExist {
-		return dirs
-	}
+func getUnitDirs(rootless bool) map[string]struct{} {
+	dirs := make(map[string]struct{}, 0)
 
-	resolvedUnitDirAdminUser := resolveUnitDirAdminUser()
-	userLevelFilter := getUserLevelFilter(resolvedUnitDirAdminUser)
-
-	if rootless {
-		systemUserDirLevel := len(strings.Split(resolvedUnitDirAdminUser, string(os.PathSeparator)))
-		nonNumericFilter := getNonNumericFilter(resolvedUnitDirAdminUser, systemUserDirLevel)
-		return getRootlessDirs(nonNumericFilter, userLevelFilter)
+	// Allow overriding source dir, this is mainly for the CI tests
+	varExist, dirs := getDirsFromEnv(dirs)
+	if !varExist {
+		resolvedUnitDirAdminUser := resolveUnitDirAdminUser()
```

Comment:
I this is would be cleaner written as `if varExist {return dirs}` so you do not have to indent the other logic

Reply:
You're right, fixed. With the addition of not returning `dirs` the `if` statement become even simpler